packages/relay/src/ServerConfigParams.ts

```diff
@@ -368,6 +370,22 @@ export function validateBalanceParams (config: ServerConfigParams): void {
   }
 }
 
+export async function validateHub (hubFile: string, hubAddress: string, managerAddress: string, workerAddress: string, contractInteractor: ContractInteractor): Promise<void> {
+  const managerNonce = await contractInteractor.getTransactionCount(managerAddress, 'pending')
+  const workerNonce = await contractInteractor.getTransactionCount(workerAddress, 'pending')
+  if (!fs.existsSync(hubFile) || (managerNonce === 0 && workerNonce === 0)) {
```

Comment:
Actually, what is the added value of having `managerNonce === 0 && workerNonce === 0` here? This is a very-very edge case if you ask me. Let's maybe have a simple, binary rule:
IF (FILE_EXISTS) {
 HUB_ADDRESS_MUST_MATCH()
} ELSE {
 CREATE_FILE()
}

Reply:
It might be the case where the dev got the hub wrong then fixed it, and if that happens after funding the manager, that would be annoying I guess. What's the downside?

Reply:
> It might be the case where the dev got the hub wrong then fixed it, and if that happens after funding the manager, that would be annoying I guess. What's the downside?

It is super edge case and handling it explicitly is unnecessary. I vote to remove this scenario. 
The downside is increased complexity. The upside is not even there as far as I can understand.

Reply:
manager funding doesn't really matter: you can always withdraw, both deposit and account balance.
its the worker we need to be careful about.
- the simple scenario to protect: avoid re-using a worker on another hub.
   - This is simply protected by never addWorker if it has balance/nonce: it was probably used in another relay (since our server only send balance after addWorker)
- the complex scenario: worker was added to a hub, but before doing anything with it the owner switched hub.
   -   this is complex, since there is no "footprint" on the worker, so @shahaf tried to resolve it by remembering (off-chain) the old hub.
   - while I agree we need shahaf's method, its not good to rely on it, so I want to use the above simple protection.
   - the needed change: transfer funds to a worker immediately after addWorker succeeds. This way, there is almost no time a worker is added but with no on-chain footprint. This is not bullet-proof (hence the need of the offchain mechanism) - but it is very close